kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,619 @@
+import org.freedesktop.dbus.types.Variant;
+import org.freedesktop.networkmanager.Device;
+import org.freedesktop.networkmanager.GetAppliedConnectionTuple;
+import org.freedesktop.networkmanager.Settings;
+import org.freedesktop.networkmanager.device.Generic;
+import org.freedesktop.networkmanager.settings.Connection;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class NMDbusConnectorTest {
+
+    DBusConnection dbusConnection = mock(DBusConnection.class, RETURNS_SMART_NULLS);
+    NetworkManager mockedNetworkManager = mock(NetworkManager.class);
+    NMDbusConnector instanceNMDbusConnector;
+    DBusConnection dbusConnectionInternal;
+
+    Boolean hasDBusExceptionBeenThrown = false;
+    Boolean hasNoSuchElementExceptionThrown = false;
+    Boolean hasNullPointerExceptionThrown = false;
+
+    NetworkInterfaceStatus netInterface;
+    NetworkService networkService;
+
+    Map<String, Device> mockDevices = new HashMap<>();
+
+    List<String> internalStringList;
+    Map<String, Object> netConfig = new HashMap<>();
+
+    @After
+    public void tearDown() {
+        resetSingleton(NMDbusConnector.class, "instance");
+    }
+
+    public static <T> void resetSingleton(Class<T> clazz, String fieldName) {
+        Field instance;
+        try {
+            instance = clazz.getDeclaredField(fieldName);
+            instance.setAccessible(true);
+            instance.set(null, null);
+            instance.setAccessible(false);
+        } catch (Exception e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Test
+    public void getDbusConnectionShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        whenGetDbusConnectionIsRun();
+
+        thenNoExceptionIsThrown();
+
+        thenGetDbusConnectionIsMockedConnection();
+    }
+
+    @Test
+    public void checkPermissionsShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedPermissions();
+
+        whenCheckPermissionsIsRun();
+
+        thenNoExceptionIsThrown();
+        thenCheckPermissionsRan();
+    }
+
+    @Test
+    public void checkVersionShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedVersion();
+
+        whenCheckVersionIsRun();
+
+        thenNoExceptionIsThrown();
+        thenCheckVersionIsRun();
+    }
+
+    @Test
+    public void getInterfacesShouldWork() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED, true);
+        givenMockedDeviceList();
+
+        whenGetInterfacesIsCalled();
+
+        thenNoExceptionIsThrown();
+        thenReturnedDevicesAre(Arrays.asList("wlan0", "eth0"));
```

Comment:
Present tense or past tense?

```suggestion
        thenGetInterfacesReturned(Arrays.asList("wlan0", "eth0"));
```